src/Confluent.Kafka/Timestamp.cs

```diff
@@ -85,12 +87,13 @@ public override int GetHashCode()
         /// </param>
         /// <returns>
         ///     The milliseconds unix timestamp corresponding to <paramref name="dateTime"/>
+        ///     rounded to the previous millisecond if dateTime precision is below millisecond
         /// </returns>
         public static long DateTimeToUnixTimestampMs(DateTime dateTime)
         {
             checked
             {
-                return (long)(dateTime.ToUniversalTime() - UnixEpochUtcDateTime).TotalMilliseconds;
+                return dateTime.ToUniversalTime().Ticks / TimeSpan.TicksPerMillisecond - UnixEpochMilliseconds;
```

Comment:
Rounding down is how microsoft do it in DateTimeOffset.ToUnixTimestamp (we could actually retake their function, but not available on net45 and would require an intermediate DateTimeOffset)

Same than you, I never except to have below millisecond precision (except when building from a DateTime.Now as it's often not rounded tick) and don't want to project in future.

I'm wondering if we should keep the DateTime constructor or simply remove it (depreciate first). It should never be used and is misleading with the datetime comparaison